website/docs/docs/build/unit-tests.md

```diff
@@ -0,0 +1,139 @@
+
+Now, we are introducing a new type of test to dbt - unit tests. In software programming, unit tests validate small portions of your functional code, and they work much the same way here. Unit tests allow you to validate your SQL modeling logic on a small set of static inputs _before_ you materialize your full model in production. Unit tests enable test-driven development, benefiting developer efficiency and code reliability. 
+
+Let’s say you’re creating a new `dim_customers` model with a field `is_valid_email_address`, that calculates whether or not the customer’s email is valid: 
+
+```sql
+with customers as (
+
+    select * from {{ ref('stg_customers') }}
+
+),
+
+accepted_email_domains as (
+
+    select * from {{ ref('top_level_email_domains') }}
+
+),
+	
+check_valid_emails as (
+
+    select
+        customers.customer_id,
+        customers.first_name,
+        customers.last_name,
+				coalesce (regexp_like(
+            customers.email, '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$'
+        )
+        = true
+        and accepted_email_domains.tld is not null,
+        false) as is_valid_email_address
+    from customers
+		left join accepted_email_domains
+        on customers.email_top_level_domain = lower(accepted_email_domains.tld)
+
+)
+
+select * from check_valid_emails
+```
+
+This type of logic can be challenging to validate. Let’s add a unit test to this model to ensure our `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
+
+```yaml
+unit_tests:
+  - name: test_is_valid_email_address
+    description: "Check my is_valid_email_address logic captures all known edge cases - emails without ., emails without @, and emails from invalid domains."
+    model: dim_customers
+    given:
+      - input: ref('stg_customers')
+        rows:
+          - {customer_id: 1, email: cool@example.com, email_top_level_domain: example.com}
+          - {customer_id: 2, email: cool@unknown.com, email_top_level_domain: unknown.com}
+          - {customer_id: 3, email: badgmail.com, email_top_level_domain: gmail.com}
+          - {customer_id: 4, email: missingdot@gmailcom, email_top_level_domain: gmail.com}
+      - input: ref('top_level_email_domains')
+        rows:
+          - {tld: example.com}
+          - {tld: gmail.com}
+    expect:
+      rows:
+        - {customer_id: 1, is_valid_email_address: true}
+        - {customer_id: 2, is_valid_email_address: false}
+        - {customer_id: 3, is_valid_email_address: false}
+        - {customer_id: 4, is_valid_email_address: false}
+
+```
+
+The above example defines the mock data using the inline `dict` format, but there are a handful of different options for how you format your mock data. 
```

Comment:
Would we want to link out to something about `format`?

Reply:
@graciegoheen did you have particular article in mind?

Reply:
Not an article - just maybe somewhere we mention the accepted values for `format`

Reply:
I think maybe just a section on the page for `format`? Similar to how there's a section for "storing test failures" on the data tests page?